dining/views.py

```diff
@@ -0,0 +1,348 @@
+
+        if frequent_locations:
+            json["cards"].update({"frequent-locations": frequent_locations})
+
+        return JsonResponse(json)
+
+    def balance(self, uid):
+        balance = DiningBalance.objects.filter(account=uid).order_by("-created_at")
+        latest = balance[0]
+        return {
+            "swipes": latest.swipes,
+            "dining-dollars": latest.dining_dollars,
+            "guest-swipes": latest.guest_swipes,
+        }
+
+    def recent_transactions_card(self, uid):
+
+        card = {
+            "type": "recent-transactions",
+            "header-title": "Transactions",
+            "header-body": "Your recent dining dollar transactions",
+            "data": [],
+        }
+
+        transactions = DiningTransaction.objects.filter(account=uid).order_by("-date")
+
+        transactions = transactions[0:5]
+
+        eastern = timezone("US/Eastern")
+
+        for transaction in transactions:
+            card["data"].append(
+                {
+                    "location": map_to_pretty(transaction.description),
+                    "date": transaction.date.replace(tzinfo=eastern).isoformat(),
+                    "balance": transaction.balance,
+                    "amount": transaction.amount,
+                }
+            )
+
+        return card
+
+    def get_semester_start_end(self):
+
+        current_start = get_value("semester_start")
+        current_end = get_value("semester_end")
+
+        try:
+            current_start = datetime.strptime(current_start, "%Y-%m-%d")
+            current_end = datetime.strptime(current_end, "%Y-%m-%d")
+        except ValueError:
+            pass
+
+        if current_start is not None and current_end is not None and current_end > datetime.now():
+            return current_start, current_end
+        else:
+            if current_start is None or current_end is None:
+                new_start_option = Option(key="semester_start", value="")
+                new_end_option = Option(key="semester_end", value="")
+            else:
+                new_start_option = get_option("semester_start")
+                new_end_option = get_option("semester_end")
+
+            new_start, new_end = get_new_start_end()
+
+            new_start_option.value = new_start.strftime("%Y-%m-%d")
+            new_start_option.save()
+
+            new_end_option.value = new_end.strftime("%Y-%m-%d")
+            new_end_option.save()
+
+            return (
+                datetime.strptime(get_value("semester_start"), "%Y-%m-%d"),
+                datetime.strptime(get_value("semester_end"), "%Y-%m-%d"),
+            )
+
+    def get_average_balances(self, uid, start_of_semester):
+
+        two_weeks_ago = datetime.now() - timedelta(days=14)
+        eastern = timezone("US/Eastern")
+        two_weeks_ago = two_weeks_ago.replace(
+            tzinfo=eastern, hour=0, minute=0, second=0, microsecond=0
```

Comment:
I would put a comment somewhere explaining why we're setting all the timezones to eastern, just so people know if they need to do it in new code they may write.